src/js_sync.hpp

```diff
@@ -404,7 +405,11 @@ void SyncClass<T>::populate_sync_config(ContextType ctx, ObjectType realm_constr
         }
 
         std::string raw_realm_url = Object::validated_get_string(ctx, sync_config_object, "url");
-
+        if (shared_user->token_type() == SyncUser::TokenType::Admin) {
+            static std::regex tilde("/~/");
+            raw_realm_url = std::regex_replace(raw_realm_url, tilde, "/__auth/");
```

Comment:
The identity is always a guid (because that’s what we provide on the JS side). I was originally thinking about using `__auth` as the identity but then there would be conflicts if you try to use an admin token user from two or more servers.